powertools-serialization/src/main/java/software/amazon/lambda/powertools/utilities/eventpart/factory/EventPartResolverFactory.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.utilities.eventpart.factory;
+
+import com.amazonaws.services.lambda.runtime.events.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.lambda.powertools.utilities.eventpart.resolvers.*;
+
+import java.util.Map;
+
+/**
+ * Factory Class that identifies the event type and invokes the appropriate implementation of an {@link EventPartResolver}
+ * <p>
+ * Main events are built-in:
+ * <ul>
+ *     <li>{@link APIGatewayProxyRequestEventPartResolver}      -> body</li>
+ *     <li>{@link APIGatewayV2HTTPEvent}            -> body</li>
+ *     <li>{@link SNSEvent}                         -> Records[0].Sns.Message</li>
+ *     <li>{@link SQSEvent}                         -> Records[*].body <i>(list)</i></li>
+ *     <li>{@link ScheduledEvent}                   -> detail</li>
+ *     <li>{@link ApplicationLoadBalancerRequestEvent} -> body</li>
+ *     <li>{@link CloudWatchLogsEvent}              -> powertools_base64_gzip(data)</li>
+ *     <li>{@link CloudFormationCustomResourceEvent} -> resourceProperties</li>
+ *     <li>{@link KinesisEvent}                     -> Records[*].kinesis.powertools_base64(data) <i>(list)</i></li>
+ *     <li>{@link KinesisFirehoseEvent}             -> Records[*].powertools_base64(data) <i>(list)</i></li>
+ *     <li>{@link KafkaEvent}                       -> records[*].values[*].powertools_base64(value) <i>(list)</i></li>
+ *     <li>{@link ActiveMQEvent}                    -> messages[*].powertools_base64(data) <i>(list)</i></li>
+ *     <li>{@link RabbitMQEvent}                    -> rmqMessagesByQueue[*].values[*].powertools_base64(data) <i>(list)</i></li>
+ *     <li>{@link KinesisAnalyticsFirehoseInputPreprocessingEvent} -> Records[*].kinesis.powertools_base64(data) <i>(list)</i></li>
+ *     <li>{@link KinesisAnalyticsStreamsInputPreprocessingEvent} > Records[*].kinesis.powertools_base64(data) <i>(list)</i></li>
+ *     <li>{@link String}</li>
+ *     <li>{@link Map}</li>
+ * </ul>
+ */
+public class EventPartResolverFactory {
```

Comment:
Nothing more than the obvious reasons, it can be better maintained, tested, maybe better understood by future devs, seperation of concerns (for each resolver), hide the actual creation logic..
But, cold starts are a really valid concern to take into account, actually I was afraid that this was the reason behind the existing implementation, but I wanted to get inputs/comments on that. 

So I will remove it and we can move on as discussed [here](https://github.com/aws-powertools/powertools-lambda-java/pull/1264#issuecomment-1621688153)

Reply:
If you can find a cleaner way without introducing X new classes/files, I'm happy too ;-) 

We also use a bit of reflection here and there, which is not good, would be great to think about it one day, but that's another story.

Reply:
ok, I'll keep both in mind. See if I can think of something